Mathlib/RingTheory/Nilpotent/Basic.lean

```diff
@@ -3,14 +3,10 @@ Copyright (c) 2021 Oliver Nash. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Oliver Nash
 -/
-import Mathlib.Algebra.Algebra.Bilinear
 import Mathlib.Algebra.GeomSum
-import Mathlib.Algebra.GroupPower.Ring
-import Mathlib.Algebra.GroupWithZero.NonZeroDivisors
-import Mathlib.Data.Nat.Choose.Sum
-import Mathlib.LinearAlgebra.Matrix.ToLin
-import Mathlib.LinearAlgebra.Quotient
-import Mathlib.RingTheory.Ideal.Operations
+import Mathlib.Algebra.Module.Basic
+import Mathlib.Algebra.SMulWithZero
+import Mathlib.Data.Nat.Lattice
 
 #align_import ring_theory.nilpotent from "leanprover-community/mathlib"@"da420a8c6dd5bdfb85c4ced85c34388f633bc6ff"
 
```

Comment:
(Can't put a comment down there but:) The module doc lists the main results, a lot of which got moved. I think it's more natural to keep those easily accessible in a light-weight file.

Perhaps it would be an option to split into the following three files:
 * definition of Nilpotent (minimal imports)
 * basic results of Nilpotent on rings (shows that nilpotent elements are closed under all the operations: the current contents of the `Lemmas.lean` file up to, but excluding, definition of `nilradical`?)
 * extra results that need the full complement of imports.

Reply:
I agree that this split is not the cleanest. But after playing around for a while I couldn't figure out a real improvement. So I think the three-way split is the way to go.